lib/cluster/instances.rb

```diff
@@ -95,9 +95,30 @@ def self.create_custom_tags
         return
       end
 
-      resource_ids = Stack.find_volume_ids
-      find_existing.each do |instance|
-        resource_ids.push(instance.ec2_instance_id)
+      vpc = VPC.find_existing
+
+      # ec2_client doesn't bring detached volumes, have to use opsworks_client
+      resource_ids = [].concat(Stack.find_volume_ids)
+
+      # opsworks_client doesn't bring the NAT instance, have to use ec2_client
+      # note that there will be duplicated volumes in the array
```

Comment:
i assume this isn't an issue when passing the list to `ec2_client.create_tags`? I'm sure ruby has some kind of unique list / set type if you're concerned.